docs/DeveloperGuide.md

```diff
@@ -1,38 +1,314 @@
+- Parses user inputs in the main session and the study session
+- Ensures the input adheres to the allowed command inputs, 
+i.e. program will ask the user to key in a new input if it does not recognise the inputs given.
+
+The `StudyCommand` component
+- Accepts user inputs in the study session
+- Facilitates interaction between Parser and TimerLogic
+
+The `TimerLogic` component
+- Manages the timer component when study session is launched
+- Handles the logic for the timer (e.g. keeps track of state of Timer component)
+- Calls made to Timer component methods are made through this method
+
+The `Timer` component
+
+![image](https://user-images.githubusercontent.com/69501969/160758578-39b9b3f3-d1ca-429d-8319-9f28e9c199c3.png)
+
+- Consists of abstract `Timer` class, `Countdown` class and `Stopwatch` class as depicted in the class diagram above
+- `Timer` inherits from Java's `Thread` class
+- `Countdown` and `Stopwatch` inherit from `Timer`
+- `Countdown` and `Stopwatch` keep track of time remaining and elapsed respectively
+- `Countdown` and `Stopwatch` can be paused, resumed and stopped
+- `Countdown` and `Stopwatch` prints to standard output the time remaining and elapsed respectively at regular intervals
+
+#### Timer implementation
+
+Sherpass’ implementation of the timer function in the study session is mainly through `Timer`, `Countdown` and 
+`Stopwatch` class. Depending on the type of timer selected by the user, either `Countdown` or `Stopwatch` will be
+instantiated. 
+
+When a countdown timer is started by the user, the `Countdown` class starts a thread which keeps track of time through
+a method called update(), where the thread sleeps for 1 second, then updates the time left, until the time left in the 
+timer reaches zero, which then interrupts the thread. Starting the timer as a thread allows us to accept user commands 
+like pause and stop for the timer through `StudyCommand` and `TimerLogic`, while the thread executes in the background 
+and prints the time remaining at regular intervals. A similar process is followed in `Stopwatch`, except that the class
+keeps track of time elapsed rather than time remaining.
+
+#### Study session usage scenario
+
+Given below is an example usage scenario when the user enters the study session, starts a countdown timer, then stops 
+the timer.
+
+Step 1. The user executes the `study` command and enters the study session through the main `Parser` component, which 
+executes the `StudyCommand`. `StudyCommand` then initialises an instance of `TimerLogic`, which handles the execution
+and logic of user commands during the study session, while the `StudyCommand` accepts the user’s input when the 
+user is in the study session. `TimerParser` parses user inputs (commands) related to `Timer`.
+
+Sequence diagram for `Timer` when user starts and stops a timer:
+
+![TimerClassSD](https://user-images.githubusercontent.com/69501969/160768104-fa7e06e3-1be8-4387-b75d-ae4e79bca5b7.png)
```

Comment:
is update command here a method? if so need to do the proper self invoke syntax, arrow itself, activate, unarrow itself, deactivate. Same as start

Reply:
![image](https://user-images.githubusercontent.com/69450377/160962525-34390ca7-2152-41ba-b362-c2839b3ce441.png)
what causes this? what happens here after the command is ran? Its abit unclear from the diagram